functions/groups/users.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * User object shape:
+ * {
+ *  role: user, admin, pending, god -> helps define permissions
+ *  cities: [city1, city2] -> Shows what cities they can moderate. Most granulate
+ *  states: [NE, IA] -> Shows what states they can moderate. Can do whole state if this is listed
+ *  change_history: Future link to all their change approves and deletes.
+ *  other_info: We might want to add other info, like name etc, but that isn't necessary to start
+ * }
+ */
+
+exports.addUser = functions.https.onRequest(async (request, response) => {
+    // PreFunctionChecks handles a bunch of CORS, verifying auth, etc.
+    // It also returns the userId if one exists base on the auth token
+    var preCheck = await helpers.PreFunctionChecks(request, response, true, false);
+    if (preCheck.ret) return;
+    
+    // Grab data from the request
+    // Data is always in the request.body.data.yourDataField sub-object
+    var userObj = request.body.data.user;
```

Comment:
How does this attach to auth?

Reply:
So, this function is only for adding the database entry for users. That is where we store their roles and any other metadata about them. We use firebase's auth api to create the actual username/password type combo. Then firebase gives that `auth user` an id which we use as our key on our `database user` does that make sense?

Reply:
Okay great. The `id` was my main question